server/cluster_info.go

```diff
@@ -446,7 +446,6 @@ func (c *clusterInfo) updateStoreStatusLocked(id uint64) {
 
 // handleRegionHeartbeat updates the region information.
 func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
-	region = region.Clone()
```

Comment:
The region is used after calling this function. Are you sure that it is safe to not clone? See: https://github.com/pingcap/pd/blob/9b0763fb0474c657c1abbef8238871f3d1284817/server/cluster_worker.go#L28-L41

Reply:
Actually, all function only read it. More reassuring I clone the region which has operator now.

Reply:
It does not make sense to clone the region. Cloning the region **reads** the region too.

I'm not so sure that the region is inmutable. I think you can propose another PR to change all fields of `core.RegionInfo` to unexported fields and add read only methods to access them. After that we can be sure that the `RegionInfo` is inmutable and discard the `Clone` method of it.

Reply:
got it.